Allow use of the native Android Debug Bridge

DanTheMan827 opened the pull request:
This allows the website to connect to the local adb server through a websocket proxy

https://github.com/dantheman827/mbf-bridge/

DanTheMan827 committed:
Allow use of the native Android Debug Bridge

This allows the website to connect to the local adb server through a websocket proxy

DanTheMan827 committed:
Don't scan for devices if a device was chosen

DanTheMan827 committed:
Update credits

DanTheMan827 committed:
Use specified bridge, and ping every 5 seconds once a device is connected

DanTheMan827 committed:
Improve device connection handling

DanTheMan827 committed:
Add OpenLogsButton to device selection and improve CSS layout for connected devices

DanTheMan827 committed:
Skip oculus check if bridge detected

DanTheMan827 committed:
Refactor logging to include raw data in console log messages for improved debugging

DanTheMan827 committed:
Implement disconnect tracking for old adb server versions

DanTheMan827 committed:
Refactor sendRequest to work with the chunks as they are sent from the pipe

DanTheMan827 committed:
Re-factor bridge data into a class

DanTheMan827 committed:
Auto-connect if bridge and only one device is found

Lauriethefish commented:
Is there a reason this PR includes making the agent support any app ID? I'd probably prefer this to be in a different PR, esp. given that the implementation of it here isn't ideal. (i.e. would prefer not to use a command line arg for the agent and would also prefer not having a bunch of different oncelocks for each static path)

DanTheMan827 committed:
Update disconnect detection logic

DanTheMan827 commented:
I removed the commit to change the game id from the agent from this branch, but I would prefer to keep it in there in some form as a dev option at least (I have two BS installs)

Lauriethefish commented on mbf-site/src/AdbServerWebSocketConnector.ts:
```diff
@@ -0,0 +1,227 @@
+              try {
+                controller.close();
+              } catch {
+                // Ignore errors during stream close.
```
Should probably be logged?

Lauriethefish commented on mbf-site/src/AdbServerWebSocketConnector.ts:
```diff
@@ -0,0 +1,227 @@
+    // If an error occurs before the socket opens, reject the openDeferred.
+    this.socket.onerror = () => {
+      if (!hasOpened) {
+        this.openDeferred.reject(new Error("WebSocket connection error"));
```
Log type of error?

Lauriethefish commented on mbf-site/src/Agent.ts:
```diff
@@ -27,7 +28,7 @@ export async function prepareAgent(adb: Adb) {
     .toUpperCase();
   Log.debug("Existing agent SHA1: " + existingSha1);
-  const existingUpToDate = AGENT_SHA1 == existingSha1.trim().toUpperCase();
+  const existingUpToDate = AGENT_SHA1 && AGENT_SHA1 == existingSha1.trim().toUpperCase();
```
Any specific reason this was added? AGENT_SHA1 should never be undefined?

Lauriethefish commented on mbf-site/src/App.tsx:
```diff
@@ -71,12 +78,126 @@ export async function getAndroidVersion(device: Adb) {
+  });
+
+  // Update the available devices on an interval
+  useEffect(() => {
```
This component is definitely getting to the size where it should be broken up a bit in my opinion.

Lauriethefish commented on mbf-site/package.json:
```diff
@@ -55,6 +54,8 @@
     ]
   },
   "devDependencies": {
+    "@yume-chan/stream-extra": "^0.0.24",
```
Oh, also, why is this still in the dependencies at all? Is there definitely a context where it's used as a dev dependency?

DanTheMan827 replied:
I saved it to the wrong list...

DanTheMan827 committed:
chore: update dependencies and improve error handling

- Moved "@yume-chan/stream-extra" from devDependencies to dependencies
- Enhanced WebSocket error handling with logging
- Log errors during stream close in WebSocketConnection
- Removed unneccesary AGENT_SHA1 empty check

DanTheMan827 committed:
Refactor code to split App component

Lauriethefish commented:
Conflicts need resolving again

DanTheMan827 committed:
Refactor device and connection management

- Removed DeviceStore and SyncStore, consolidating their functionality into a new context-based approach.
- Introduce DeviceConnector and BridgeManager hooks to manage device connections and bridge interactions.
- Updated ModManager, OptionsMenu, and other components to utilize the new context for device management.
- Refactor operation modals to use a new context-based approach.
- Add a missing key in the ChooseDevice component.

DanTheMan827 committed:
Abstract bridge interface to allow for different bridges

DanTheMan827 committed:
Implement __mbfBridge ADB connector

This uses a custom `__mbfBridge` window object injected by a wrapper app to connect to the native adb server running on the device.

DanTheMan827 commented:
@Lauriethefish I’ve abstracted the ADB connector logic, and also added support for a global `__mbfBridge` object that can be implemented by apps (and potentially even browser extensions) to allow for connection to the ADB server on the host machine without needing to create a web socket.

I’ve kept the web socket connector in this branch, but it could just as well be removed.  The factory method allows for multiple potential connectors (or just one)
